services/dashboard/routes/workers.go

```diff
@@ -0,0 +1,153 @@
+package routes
+
+import (
+	"encoding/base64"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+
+	"sensorbucket.nl/sensorbucket/internal/web"
+	"sensorbucket.nl/sensorbucket/pkg/api"
+	"sensorbucket.nl/sensorbucket/services/dashboard/views"
+)
+
+type WorkerPageHandler struct {
+	router chi.Router
+	client *api.APIClient
+}
+
+func CreateWorkerPageHandler(client *api.APIClient) *WorkerPageHandler {
+	handler := &WorkerPageHandler{
+		router: chi.NewRouter(),
+		client: client,
+	}
+	handler.SetupRoutes(handler.router)
+	return handler
+}
+
+func (h WorkerPageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.router.ServeHTTP(w, r)
+}
+
+func (h *WorkerPageHandler) SetupRoutes(r chi.Router) {
+	r.Get("/", h.listWorkers())
+	r.Get("/create", h.createWorkerPage())
+	r.Post("/create", h.createWorker())
+	r.Get("/{id}", h.workerDetails())
+	r.Patch("/{id}", h.updateWorker())
+}
+
+func (h *WorkerPageHandler) listWorkers() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		page := &views.WorkerListPage{}
+		res, _, err := h.client.WorkersApi.ListWorkers(r.Context()).Execute()
+		if err != nil {
+			web.HTTPError(w, err)
+			return
+		}
+		page.Workers = res.Data
+
+		if isHX(r) {
+			page.WriteBody(w)
+			return
+		}
+		views.WriteIndex(w, page)
+	}
+}
+
+func (h *WorkerPageHandler) workerDetails() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		page := &views.WorkerEditorPage{}
+		workerID := chi.URLParam(r, "id")
+		res, _, err := h.client.WorkersApi.GetWorker(r.Context(), workerID).Execute()
+		if err != nil {
+			web.HTTPError(w, err)
+			return
+		}
+		resUC, _, err := h.client.WorkersApi.GetWorkerUserCode(r.Context(), workerID).Execute()
```

Comment:
is a check on the response relevant? instead of discarding it there may be an unexpected status code?

Reply:
If the status code is not 200 through 299, an error will be produced. The second return value is purely the full HTTP response, but the library already converts it into either an error or the requested entities.